### PR TITLE
parse middle_module from message type in ros2topic

### PR DESCRIPTION
**Pull request by davidhodo**
Attempts to solve issue #277.  The middle module name is parsed from the message type rather than being hardcoded as .msg.  I have tested it with a few message types generated from .msg and from .idl but not exhaustively.  Is there a case where the message_type split could result in more than 3 strings?

### CODE REVIEW

**Comment by jacobperron** [comment]
It's undecided if we'll allow message types with more than three parts. There's an open proposal that there could be more than three: https://github.com/ros2/design/pull/220
Until it is decided, I don't think we need to worry about supporting that use-case.

**Comment by davidhodo** [comment]
Ok.  I updated the request to default to 'msg' if the type only contains two parts.  Can this be merged or is it a wontfix in light of the planned new features listed in #223 mentioned by Dirk in #277?

File: ros2topic/ros2topic/api/__init__.py [hunk 1]
```diff
@@ -57,19 +57,20 @@ def __call__(self, prefix, parsed_args, **kwargs):
 def import_message_type(topic_name, message_type):
     # TODO(dirk-thomas) this logic should come from a rosidl related package
     try:
-        package_name, *message_name = message_type.split('/')
-        if not package_name or not message_name or not all(message_name):
+        package_name, middle_module, message_name = message_type.split('/')
+        if not middle_module:
```
Comment (dirk-thomas) [inline ros2topic/ros2topic/api/__init__.py]: This logic doesn't seem to make any sense. In which case would `not middle_module` be `True`?

File: ros2topic/ros2topic/api/__init__.py [hunk 2]
```diff
@@ -57,19 +57,20 @@ def __call__(self, prefix, parsed_args, **kwargs):
...
             raise ValueError()
     except ValueError:
         raise RuntimeError('The passed message type is invalid')
 
     # TODO(sloretz) node API to get topic types should indicate if action or msg
-    middle_module = 'msg'
     if topic_name.endswith('/_action/feedback'):
         middle_module = 'action'
 
     module = importlib.import_module(package_name + '.' + middle_module)
```
Comment (jacobperron) [inline ros2topic/ros2topic/api/__init__.py]: Perhaps we should just support any number of parts, e.g. `'.'.join([package_name] + message_name[:-1])`.

Comment (Karsten1987) [reply ros2topic/ros2topic/api/__init__.py]: Does it not generally make sense to unify this logic and have it in a central place? I know of at least one more other location where we duplicate the same logic:
https://github.com/ros2/ros2cli/blob/master/ros2service/ros2service/verb/call.py#L61-L72

Comment (jacobperron) [reply ros2topic/ros2topic/api/__init__.py]: Yes. There's a TODO at the top of this function. I'm not sure what package it should belong in (or if a new package needs to be created.

**Comment by nuclearsandwich** [comment]
@davidhodo will you be able to address the feedback on this PR?

**Comment by davidhodo** [comment]
Yes. Sorry for the long delay.  I will try to get it done this week.

**Comment by nuclearsandwich** [comment]
@davidhodo are you still interested in updating this PR?

**Comment by jacobperron** [comment]
I think this PR is obsolete by #415.

I'll check if #277 is resolved by #415 and if not, we should probably target a PR against https://github.com/ros2/rosidl_runtime_py.

**Comment by jacobperron** [comment]
If we want this fix for Dashing, then I suggest re-opening this PR and changing the base branch to `dashing`. Trying to backport #415 and it's dependencies is much more work.

**Comment by davidhodo** [comment]
Sorry.  I am still interested, but have been swamped and haven't been able to get back to it.  I'll rebase on Dashing and re-open as soon as I get some time.

**Pull request closed without merging**